Error when trying to update a Java code with a switch (Java 17)

### Description of the bug

If I'm running my application in dev mode and try to change the style of a code that contains a switch expression (https://docs.oracle.com/en/java/javase/17/language/switch-expressions.html#GUID-BA4F63E3-4823-43C6-A5F3-BAA4A2EF3ADC)

I have an error:
``` 
Caused by: com.github.javaparser.ParseProblemException: (line 37,col 16) Switch expressions are not supported.

	at com.github.javaparser.JavaParserAdapter.handleResult(JavaParserAdapter.java:83) ~[javaparser-core-3.25.5.jar:na]
	at com.github.javaparser.JavaParserAdapter.parse(JavaParserAdapter.java:107) ~[javaparser-core-3.25.5.jar:na]
	at com.github.javaparser.StaticJavaParser.parse(StaticJavaParser.java:253) ~[javaparser-core-3.25.5.jar:na]
	at com.vaadin.base.devserver.editor.Editor.parseSource(Editor.java:1100) ~[vaadin-dev-server-24.2.0.beta2.jar:na]
	at com.vaadin.base.devserver.editor.Editor.modifyClass(Editor.java:1038) ~[vaadin-dev-server-24.2.0.beta2.jar:na]
	at com.vaadin.base.devserver.themeeditor.JavaSourceModifier.setLocalClassName(JavaSourceModifier.java:91) ~[vaadin-dev-server-24.2.0.beta2.jar:na]
	at com.vaadin.base.devserver.themeeditor.JavaSourceModifier.lambda$setLocalClassName$25aa656f$1(JavaSourceModifier.java:78) ~[vaadin-dev-server-24.2.0.beta2.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[na:na]
	at com.vaadin.flow.server.VaadinService.runPendingAccessTasks(VaadinService.java:2018) ~[flow-server-24.2.0.beta2.jar:24.2.0.beta2]

```

### Expected behavior

I would like to be able to use all the Java 17 features by default since Java 17 is required for Vaadin 24 and also be able to use Java 21 if my application is using Java 21.

### Minimal reproducible example

https://github.com/jcgueriaud1/test-theme-editor-java17



### Versions

- Vaadin / Flow version: 24.2.0.rc2
- Java version: 17 or more

--- a/vaadin-dev-server/src/main/java/com/vaadin/base/devserver/editor/Editor.java
+++ b/vaadin-dev-server/src/main/java/com/vaadin/base/devserver/editor/Editor.java
@@ -1,5 +1,6 @@
 package com.vaadin.base.devserver.editor;
 
+import com.github.javaparser.ParserConfiguration;
 import com.github.javaparser.Position;
 import com.github.javaparser.Range;
 import com.github.javaparser.StaticJavaParser;
@@ -33,6 +34,7 @@ import com.github.javaparser.ast.stmt.ExpressionStmt;
 import com.github.javaparser.ast.stmt.Statement;
 import com.github.javaparser.ast.type.ClassOrInterfaceType;
 import com.github.javaparser.printer.lexicalpreservation.LexicalPreservingPrinter;
+import com.github.javaparser.utils.SourceRoot;
 import com.vaadin.base.devserver.themeeditor.utils.StatementLineNumberVisitor;
 import com.vaadin.flow.shared.util.SharedUtil;
 import org.apache.commons.io.IOUtils;
@@ -1097,6 +1099,10 @@ public class Editor {
     }
 
     protected CompilationUnit parseSource(String source) {
+        ParserConfiguration parserConfiguration = new ParserConfiguration();
+        parserConfiguration
+                .setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_17);
+        StaticJavaParser.setConfiguration(parserConfiguration);
         return LexicalPreservingPrinter.setup(StaticJavaParser.parse(source));
     }
 

--- a/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/editor/AddComponentTest.java
+++ b/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/editor/AddComponentTest.java
@@ -123,6 +123,7 @@ public class AddComponentTest extends AbstractClassBasedTest {
                 "Button helloWorld = new Button(\"Hello world\");");
         assertTestFileContains("add(helloWorld)");
     }
+
     @Test
     public void addTwiceToEmptyView() throws Exception {
         setupTestClass("EmptyView");

--- a/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/editor/AddComponentTest.java
+++ b/vaadin-dev-server/src/test/java/com/vaadin/base/devserver/editor/AddComponentTest.java
@@ -105,6 +105,24 @@ public class AddComponentTest extends AbstractClassBasedTest {
         assertTestFileContains("add(helloWorld)");
     }
 
+    /**
+     * Test if the Switch expression is supported (Java 17 support)
+     *
+     * @throws Exception
+     */
+    @Test
+    public void addToEmptyViewWithSwitch() throws Exception {
+        setupTestClass("EmptyViewWithSwitch");
+        // Create is the constructor
+        int create = getLineNumber(testFile, "public EmptyView() {");
+        int attach = create;
+        editor.addComponent(testFile, create, attach, Where.INSIDE,
+                ComponentType.BUTTON, "Hello world");
+
+        assertTestFileContains(
+                "Button helloWorld = new Button(\"Hello world\");");
+        assertTestFileContains("add(helloWorld)");
+    }
     @Test
     public void addTwiceToEmptyView() throws Exception {
         setupTestClass("EmptyView");

--- a/vaadin-dev-server/src/test/resources/com/vaadin/base/devserver/editor/inputs/EmptyViewWithSwitch.java
+++ b/vaadin-dev-server/src/test/resources/com/vaadin/base/devserver/editor/inputs/EmptyViewWithSwitch.java
@@ -1,0 +1,20 @@
+package com.vaadin.base.devserver.editor.inputs;
+
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+
+@Route(value = "empty")
+public class EmptyView extends HorizontalLayout {
+
+    public EmptyView() {
+    }
+
+    private String testSwitch(String value) {
+        return switch (value) {
+            case "test" -> "test";
+            case "prod" -> "Production";
+            default -> "N/A";
+        };
+    }
+}

